### PR TITLE
Build arrow from source on Linux

### PR DESCRIPTION
**Pull request by SimbaGithub**
Simba ticket 00332068
To support ARM Linux, build arrow from source code on Linux.

### CODE REVIEW

File: scripts/build_arrow_linux.sh [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+source $DIR/utils.sh
+
+[[ -n "$GET_VERSION" ]] && echo $AZURE_VERSION && exit 0
+
+ARROW_SOURCE_DIR=$DEPS_DIR/arrow
+ARROW_BUILD_DIR=$DEPENDENCY_DIR/arrow
+ARROW_DEPS_BUILD_DIR=$DEPENDENCY_DIR/arrow_deps
+ARROW_CMAKE_BUILD_DIR=$ARROW_SOURCE_DIR/cpp/cmake-build
+
+GIT_REPO="https://github.com/apache/arrow.git"
```
Comment (sfc-gh-cshi) [inline scripts/build_arrow_linux.sh:35]: Not sure if we should fetch arrow directly from here, it is not a small fetch. Let me check if we should use some cache here.

File: scripts/build_arrow_linux.sh [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+#
+function usage() {
+    echo "Usage: `basename $0` [-t <Release|Debug>]"
+    echo "Build ARROW"
+    echo "-t <Release/Debug> : Release or Debug builds"
+    exit 2
+}
+set -o pipefail
+
+ARROW_VERSION=0.17.1
```
[the remark this answers is not in the document]
Comment (SimbaGithub) [reply scripts/build_arrow_linux.sh:13]: > what is the reason we picked this version?

That's the latest version and we were on this version before I adding the scripts of building arrow from source.

**Comment by sfc-gh-cshi** [comment]
I got a segmentation fault when running ci/build_linux.sh locally, the same as in the "@github-actions
Build and Test / Build-Test-Linux (Release, AZURE) (pull_request)". Please solve the pre-commit failures.

**Review by sfc-gh-cshi — APPROVED** [review]

**Comment by SimbaGithub** [comment]
@sfc-gh-cshi confirmed XP build went well.

**Pull request merged**